sphinx/themes/basic/static/searchtools.js

```diff
@@ -510,9 +511,27 @@ const Search = {
       if (word.length > 2) {
         const escapedWord = _escapeRegExp(word);
         if (!terms.hasOwnProperty(word)) {
-          Object.keys(terms).forEach((term) => {
-            if (term.match(escapedWord))
-              arr.push({ files: terms[term], score: Scorer.partialTerm });
+          const termOffsets = Object.keys(terms);
+          const ngramTerms = function (ngram) {
+            let [node, path] = [termsNgrams, ""];
+            for (const step of ngram) {
+              if ((path += step) in node) [node, path] = [node[path], ""];
+            }
+            if (!node) return [];
+            if (node.length === undefined) node = [node];
+            return node.map(offset => termOffsets[offset]);
```

Comment:
I think there's still a problematic codeflow somewhere in here; in particular I think if the `node` value ends up being somewhere intermediate in the tree (an `Object` -- one of the nodes in the tree; that is, not a singleton / `int` value, and not an `Array` of `int` values either) then the result is `undefined`.

Todo: add some covering tests and a fix for that.

Reply:
Resolved by returning an empty list when traversing the trie nodes does not completely consume the path (the substring being matched) in commit 94daefdefb9caf916d401920e8e930c88fbf0a9c.

Existing test coverage - in particular [`should handle queries with no matches`](https://github.com/jayaddison/sphinx/blob/d38500be94a466cdc7885eda66fbcf547d59b23b/tests/js/searchtools.js#L202-L207) - exercises this condition.